internal/xds/hash.go

```diff
@@ -14,23 +14,32 @@
 package xds
 
 import (
-	envoy_api_v2_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
-	cache "github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+	envoy_api_core_v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	envoy_config_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 )
 
-// ConstantHash is a specialized node ID hasher used to allow
+const CONSTANT_HASH_VALUE = "contour"
+
+// ConstantHashV2 is a specialized node ID hasher used to allow
 // any instance of Envoy to connect to Contour regardless of the
 // service-node flag configured on Envoy.
-type ConstantHash string
+type ConstantHashV2 struct{}
```

Comment:
is it worth moving this into `internal/xds/v2`, and the v3 bits into `internal/xds/v3`?

Reply:
I think that I'd rather leave it for now. If I move the 2 implementations I have to either duplicate the "contour" value, or leave the `CONSTANT_HASH_VALUE` const behind. I don't mind moving them, but I didn't like either of those options very much.